_posts/2021-04-22-release-1.13.0.md

```diff
@@ -0,0 +1,483 @@
+```
+Note that Hive dialect no longer supports Flink syntax for DML and DQL. Switch to `default` dialect
+when you need to write Flink syntax.
+
+**Batch support in PyFlink DataStream API**
+
+In 1.12, it has added batch execution mode in the Java DataStream API. In this release, this feature
+is also added in the Python DataStream API.
+
+**General Python User-Defined Aggregate Function Support in group window of PyFlink Table API**
+
+It supported both general Python UDAF and pandas UDAF in PyFlink Table API. In the previous releases,
+The general Python UDAF could only be used in unbounded groupby aggregation. In this release,
+it's also supported to be used in group window of PyFlink Table API.
+
+**Improved performance of Sort-Merge Shuffles**
+
+In 1.12, we introduced sort-merge blocking shuffle to Flink (FLIP-148). In this release the feature
+was optimized to improve usability (fixed direct memory OOM issue) and performance (introduced IO
+scheduling and broadcast optimization).
+
+**HBase connector supports async lookup and lookup cache**
+
+Before 1.13, HBase lookup source can only be used in sync mode which means the communication between
+Flink and HBase is synchronous. This leads to poor throughput performance. In 1.13, we introduced
+async lookup mode for HBase connector and also a lookup cache to reduce I/O interaction. This can
+make Table/SQL jobs have much better throughput when lookup joining HBase.
+
+## Important Changes
+
+* [FLINK-21709](https://issues.apache.org/jira/browse/FLINK-21709) - The old planner of the Table &
+  SQL API has been deprecated in Flink 1.13 and will be dropped in Flink 1.14. This means that both
+  the BatchTableEnvironment and DataSet API interop are reaching end of life. You should use the
+  unified TableEnvironment for batch and stream processing with the new planner, or the DataStream
+  API in batch execution mode.
+* [FLINK-21935](https://issues.apache.org/jira/browse/FLINK-21935) - The `state.backend.async`
+  option is deprecated. Snapshots are always asynchronous now (as they were by default before) and
+  there is no option to configure a synchronous snapshot anymore.
+* [FLINK-17012](https://issues.apache.org/jira/browse/FLINK-17012) - Task's RUNNING state was split
+  into two states: RECOVERING and RUNNING. A task is RECOVERING while the state is initialising and,
+  in the case of unaligned checkpoints, until all of the in-flight data has been recovered.
+* [FLINK-21698](https://issues.apache.org/jira/browse/FLINK-21698) - The CAST operation between
+  NUMERIC type and TIMESTAMP type is problematic and is disabled now, e.g. CAST(numeric AS TIMESTAMP(3))
+  is disabled. You should use TO_TIMESTAMP(FROM_UNIXTIME(numeric)) instead.
+* [FLINK-21193](https://issues.apache.org/jira/browse/FLINK-21193) - The Flink documentation has
+  been migrated from Jekyll to Hugo. If you find something missing, let us know. We are also happy
+  to hear if you like the new look & feel.
+
```

Comment:
Shall we also mention the deprecation of Mesos here?

Reply:
Yes, makes total sense!